pkg/server/api/entry/v1/service.go

```diff
@@ -132,3 +165,62 @@ func applyMask(e *types.Entry, mask *types.EntryMask) { //nolint: unused,deadcod
 		e.DnsNames = nil
 	}
 }
+
+func buildListEntriesRequest(req *entry.ListEntriesRequest) (*datastore.ListRegistrationEntriesRequest, error) {
+	listReq := &datastore.ListRegistrationEntriesRequest{}
+
+	if req.PageSize > 0 {
+		listReq.Pagination = &datastore.Pagination{
+			PageSize: req.PageSize,
+			Token:    req.PageToken,
+		}
+	}
+
+	if req.Filter == nil {
+		return listReq, nil
+	}
+
+	if req.Filter.ByParentId != nil {
+		var err error
+		listReq.ByParentId, err = spiffeIDToStringValue(req.Filter.ByParentId)
+		if err != nil {
+			return nil, fmt.Errorf("ByParentId argument is not a valid SPIFFE ID: %s", req.Filter.ByParentId) //nolint: golint
+		}
+	}
+
+	if req.Filter.BySpiffeId != nil {
+		var err error
+		listReq.BySpiffeId, err = spiffeIDToStringValue(req.Filter.BySpiffeId)
+		if err != nil {
+			return nil, fmt.Errorf("BySpiffeId argument is not a valid SPIFFE ID: %s", req.Filter.BySpiffeId) //nolint: golint
+		}
+	}
+
+	if req.Filter.BySelectors != nil {
+		dsSelectors := []*common.Selector{}
+		for _, reqSelector := range req.Filter.BySelectors.Selectors {
```

Comment:
on Create issue we are adding logic to parse and validate selectors, maybe you can reuse that?

Reply:
I will into it.